src/test/java/org/commcare/formplayer/tests/CaseClaimTests.java

```diff
@@ -318,6 +267,46 @@ public void testQueryScreen() throws Exception {
         assertArrayEquals(new String[]{"False"}, requestData.get("include_closed").toArray());
     }
 
+    @Test
+    public void testDependentItemsets_DependentChoicesChangeWithSelection() throws Exception {
+        Hashtable<String, String> inputs = new Hashtable<>();
+        inputs.put("state", "1");
+        QueryData queryData = setUpQueryDataWithInput(inputs, true, false);
+        QueryResponseBean queryResponseBean = runQuery(queryData);
+        assert queryResponseBean.getDisplays()[1].getValue().contentEquals("1");
+        assertArrayEquals(queryResponseBean.getDisplays()[1].getItemsetChoices(),
+                new String[]{"karnataka", "Raj as than"});
+        assertArrayEquals(queryResponseBean.getDisplays()[2].getItemsetChoices(),
+                new String[]{"Baran", "Kota"});
+
+        inputs.put("state", "0");
+        queryResponseBean = runQuery(queryData);
+        assert queryResponseBean.getDisplays()[1].getValue().contentEquals("0");
+        assertArrayEquals(queryResponseBean.getDisplays()[1].getItemsetChoices(),
+                new String[]{"karnataka", "Raj as than"});
+        // check if we have districts corresponding to karnataka state
+        assertArrayEquals(queryResponseBean.getDisplays()[2].getItemsetChoices(),
+                new String[]{"Bangalore", "Hampi"});
+    }
+
+    @Test
+    public void testDependentItemsets_SelectionPeristsInResponse() throws Exception {
```

Comment:
This is the only real addition which addresses coverage for https://github.com/dimagi/commcare-core/pull/1142